server/models/Document.ts

```diff
@@ -436,32 +440,53 @@ class Document extends ParanoidModel {
     query: string,
     options: SearchOptions = {}
   ): Promise<SearchResponse> {
-    const limit = options.limit || 15;
-    const offset = options.offset || 0;
     const wildcardQuery = `${escape(query)}:*`;
-    const collectionIds = await team.collectionIds();
+    const { minWords = 20, maxWords = 30, limit = 15, offset = 0 } = options;
 
-    // If the team has access no public collections then shortcircuit the rest of this
+    // restrict to specific collection if provided
+    // enables search in private collections if specified
+    let collectionIds;
+    if (options.collectionId) {
+      collectionIds = [options.collectionId];
+    } else {
+      collectionIds = await team.collectionIds();
+    }
+
+    // short circuit if no relevant collections
     if (!collectionIds.length) {
       return {
         results: [],
         totalCount: 0,
       };
     }
 
-    // Build the SQL query to get documentIds, ranking, and search term context
+    // restrict to documents in the tree of a shared document when one is provided
+    let documentIds;
+
+    if (options.share?.includeChildDocuments) {
+      const sharedDocument = await options.share.$get("document");
+      if (sharedDocument) {
+        const childDocumentIds = await sharedDocument.getChildDocumentIds();
+        documentIds = [sharedDocument.id, ...childDocumentIds];
+      }
+    }
+
+    const documentClause = documentIds ? `"id" IN(:documentIds) AND` : "";
+
+    // Build the SQL query to get result documentIds, ranking, and search term context
     const whereClause = `
   "searchVector" @@ to_tsquery('english', :query) AND
     "teamId" = :teamId AND
     "collectionId" IN(:collectionIds) AND
+    ${documentClause}
     "deletedAt" IS NULL AND
     "publishedAt" IS NOT NULL
   `;
     const selectSql = `
     SELECT
       id,
       ts_rank(documents."searchVector", to_tsquery('english', :query)) as "searchRanking",
-      ts_headline('english', "text", to_tsquery('english', :query), 'MaxFragments=1, MinWords=20, MaxWords=30') as "searchContext"
+      ts_headline('english', "text", to_tsquery('english', :query), 'MaxFragments=1, MinWords=${minWords}, MaxWords=${maxWords}') as "searchContext"
```

Comment:
```suggestion
      ts_headline('english', "text", to_tsquery('english', :query), 'MaxFragments=1, MinWords=:minWords, MaxWords=:maxWords') as "searchContext"
```

These should use query replacement